src/Lean/Elab/Tactic/Simp.lean

```diff
@@ -126,11 +126,11 @@ inductive ResolveSimpIdResult where
   | ext  (ext : SimpExtension)
 
 /--
-  Elaborate extra simp theorems provided to `simp`. `stx` is of the `simpTheorem,*`
+  Elaborate extra simp theorems provided to `simp`. `stx` is of the form `"[" simpTheorem,* "]"`
```

Comment:
Suggestion: let's introduce
```
syntax simpEntry := simpStar <|> simpErase <|> simpLemma
syntax simpEntries := "[" simpEntry,* "]"
```
for better reuse and proper `TSyntax` signature. Happy to implement that if others agree.

Reply:
A few caveats: sometimes the grammar is just `"[" (simpErase <|> simpLemma),* "]"`

Reply:
Good to know, so the function should accept ``TSyntaxArray `simpEntry``

Reply:
I had a go at implementing but it got errors